### ISSUE TITLE
> This is really a great job, and thank you so much for sharing your source code. Here I have a question.

### ISSUE THREAD

**Issue by ShanZard** [songtaohe/Sat2Graph#15]
> This is really a great job, and thank you so much for sharing your source code. Here I have a question.
> If I want to try Sat2Graph on my datasets, how to obtain the sample points (_refine_gt_graph_samplepoints.json) and the neighbors (_refine_gt_graph.p) from the ground-truth (_gt.png) please?
> Appreciate your help!

In the current implementation, we take the ground truth graph (from OpenStreetMap) as input (in graph format) and generate the corresponding segmentation mask (_gt.png), the sample points (_refine_gt_graph_samplepoints.json), and the interpolated ground truth graphs (_refine_gt_graph.p). For this part, you can check the code in prepare_dataset/download.py 

If your ground-truth is in segmentation format, then you may have to first convert it to graph format. Unfortunately, there is no code in this repo. I can try to add one if you need it.   

The code to create the sample points and the refined ground truth graphs (_refine_gt_graph.p).

_Originally posted by @songtaohe in https://github.com/songtaohe/Sat2Graph/issues/2#issuecomment-709553552_

**Comment by ShanZard** [comment]
Thanks for your code. In issues 2 you reply that you will add code for ground-truth convert to graph format, so how can I find it?
感谢您的工作，我观察到您似乎也可以用中文交流，所以我将我的问题用中文说一下。我看到了apls指标的代码，然后他需要.p这种格式的文件，然而我现在只有真实的label和预测出的label并且都是.png这种图片格式的。我如何转换到.p格式呢？我在issues2 看见了您的回复，在download.py下面可以做到，但是那个代码似乎是针对SpaceNet数据集的。里面有许多的函数是针对那个数据集设计的，我似乎没有将他更改，同时您在issues 2中回复到会添加一个将分割ground-truth转换到graph format，我在哪里可以找到这段代码呢？
期待您的回复

**Comment by ShanZard** [comment]
如果您可以提供一个在其他数据集上面计算APLS指标的代码，那真的太感谢了！

**Comment by shan3960** [comment]
First off let me thank you for the Its really a great work and the source code.

I need to implement  roads extraction feature for one of my project for that we need to train the model with our own data.
As you said, there is no code available in the repo for ground-truth data conversion( segmentation format to graph format)
kindly share the training data conversion code for the same, so that we could train Sat2Graph network model with our own data set. you may mail me to my email id:  kshankar@cdac.in or shan_rathod@ymail.com

Thank you
Shankar Naik Rathod Karamtoth
Principal Technical Officer, C-DAC, Pune.
